src/edu/stanford/nlp/util/StringUtils.java

```diff
@@ -1738,13 +1739,16 @@ public static String toTitleCase(String s) {
    *         false otherwise
    */
   public static boolean isTitleCase(String s) {
-    String[] words = s.split("\\s+");
-    for (String w : words) {
-      if (!(Character.isUpperCase(w.charAt(0)) && (w.substring(1).toLowerCase().equals(w.substring(1))))) {
-        return false;
-      }
+    Predicate<String> capitalized = word -> Character.isUpperCase(word.charAt(0));
+    Predicate<String> remainderIsLowerCase = w -> w.substring(1).toLowerCase().equals(w.substring(1));
+
+    Predicate<String> titleCased = capitalized.and(remainderIsLowerCase);
```

Comment:
Let me know if I got this right, so the capitalized  and remainderIsLowerCase are like methods which are merged now in titleCased, and then the assesment is happenign in the .allMatch(titlecase) method

Reply:
Yes, you understood it absolutely right! It basically does the same thing as the previous code did - but in my opinion more readable and for very large strings it has better performance